pkg/detector/policy.go

```diff
@@ -175,6 +176,90 @@ func (d *ResourceDetector) getAndApplyClusterPolicy(object *unstructured.Unstruc
+	// this logic is also intended to be compatible with previous versions which talked about the issue of Ingress(networking.k8s.io/v1) and Ingress(extensions/v1beta1).
+	if rb.Spec.Resource.APIVersion != resource.GetAPIVersion() || rb.Spec.Resource.Kind != resource.GetKind() ||
+		rb.Spec.Resource.Name != resource.GetName() || rb.Spec.Resource.Namespace != resource.GetNamespace() {
+		klog.Errorf("The rb (%s) related resource is not match to the given resource: %s/%s", rbName, resource.GetNamespace(), resource.GetName())
+		return nil
+	}
+
+	// 3. clean the labels in binding
+	bindingCopy := rb.DeepCopy()
+	for _, label := range removeLabels {
+		delete(bindingCopy.Labels, label)
+	}
+	if err := d.Client.Update(context.TODO(), bindingCopy); err != nil {
+		klog.Errorf("Failed to remove binding labels, %s/%s, err: %+v", bindingCopy.GetNamespace(), bindingCopy.GetName(), err)
+		return err
+	}
+
+	// 4. clean the labels in resource
+	resourceCopy := resource.DeepCopy()
+	util.RemoveLabels(resourceCopy, removeLabels...)
+	if err := d.Client.Update(context.TODO(), resourceCopy); err != nil {
+		klog.Errorf("Failed to remove resource labels, %s/%s, err: %+v", resourceCopy.GetNamespace(), resourceCopy.GetName(), err)
+		return err
+	}
+
+	klog.Infof("Successfully removed the policy labels in resource (%s/%s) and rb (%s) for policy (%s) no longer matched",
+		resource.GetNamespace(), resource.GetName(), rbName, resource.GetLabels()[policyv1alpha1.PropagationPolicyNameLabel])
+	return nil
+}
+
+// removeSpecificResourceAndCRBLabels remove the cluster propagation policy labels of specific resource and corresponding cluster binding.
+func (d *ResourceDetector) removeSpecificResourceAndCRBLabels(resource *unstructured.Unstructured, removeLabels []string) error {
+	// 1. get crb of given resource
+	crbName := names.GenerateBindingName(resource.GetKind(), resource.GetName())
+	crb := &workv1alpha2.ClusterResourceBinding{}
+	if err := d.Client.Get(context.TODO(), client.ObjectKey{Name: crbName}, crb); err != nil {
+		klog.Errorf("Failed to get ClusterResourceBinding: %s, err: %+v", crbName, err)
+		return err
+	}
+
+	// 2. check whether crb related resource match to the given resource
+	// the mismatch would never happen under normal circumstances, if it really happened, just print logs and return nil.
+	// this logic is also intended to be compatible with previous versions which talked about the issue of Ingress(networking.k8s.io/v1) and Ingress(extensions/v1beta1).
+	if crb.Spec.Resource.APIVersion != resource.GetAPIVersion() || crb.Spec.Resource.Kind != resource.GetKind() ||
+		crb.Spec.Resource.Name != resource.GetName() {
+		klog.Errorf("The crb (%s) related resource is not match to the given resource: %+v/%s", crbName, resource.GroupVersionKind(), resource.GetName())
+		return nil
+	}
+
+	// 3. clean the labels in binding
+	bindingCopy := crb.DeepCopy()
+	for _, label := range removeLabels {
```

Comment:
```suggestion
	util.RemoveLabels(bindingCopy, removeLabels...)
```

Reply:
Are you sure ???

pay attention to the function declaration func RemoveLabels(obj *unstructured.Unstructured, labelKeys ...string)